### DISCUSSION TITLE
Advanced customizations

### DISCUSSION THREAD

**Discussion by fredguth** [quarto-dev/quarto-cli#2939] [Q&A] [answered]
I am trying to create a Project Type Thesis with the design of [my dissertation](https://teodecampos.github.io/fred_guth/guth_msc_unb_final_2022.pdf).  It would generate a pdf but also a site.  It resembles Tufte books layout, with some tweaks.  For my dissertation,  I used memoir and lots of custom commands, but for this Project Type, I wish to change as little as possible from Quarto defaults, while achieving a similar format (I am still unsure if I shouldn't change the documentclass). My work-in-progress can be found at  [fredguth/thesis](https://github.com/fredguth/thesis): 

1. **Margin reference style**
 - I set `reference-location: margin` and `citation-location: margin` (why the location for footnotes is `reference-location` and not `footnote-location`? ).  I noticed that quarto inserts the references in the margin with something like:
 
```latex
\marginpar{
\begin{footnotesize}\leavevmode\vadjust 
pre{\protect\hypertarget{ref-rahimi:2017}{}}%Rahimi, Ali. 2018. {``Ali Rahimi NIPS 2017 Test-of-Time Award
Presentation Speech.''} \url{https://youtu.be/x7psGHgatGM}; Youtube.
\url{https://youtu.be/x7psGHgatGM}.\vspace{2mm}
\par\end{footnotesize}}
```
How do I customize that?  I don't want the full reference in the side (I will keep full citation only in the References chapter, with a link from the margin note to the full reference).

2. **Header, section and chapter styles** 
I have tried changing the styles of section headings and chapter styles without success. I tried creating in-header.tex with things like:
````in-header.tex
\renewcommand*{\chapterformat}{%
  \thechapter\enskip
  \textcolor{gray!50}{\rule[-\dp\strutbox]{2pt}{\baselineskip}}\enskip
}
````
 - Quarto produced a header with the current chapter or current section name, but also with section numbers, but I don't want to use the chapter or section number in the header, how do I customize that? In my dissertation I used `createplainmark` but I couldn't find what was similar in the .tex generated by quarto.
 - The header is aligned with the main body, but I want to align the header in the right page with the end of `marginpar`.  
 - I want the page numbers in the header

3. **User options**
- I know how to get metadata and use in a partial, but is it possible to use it in the _extension.yaml?
I wanted the user to config something like:
```_quarto.yml
project:
    type: thesis
thesis:
    frontmatter:
        - cover.qmd
        - title.qmd
        - signatures.qmd
        - catalog.qmd
        - dedication.qmd
        - acknowledges.qmd
    matter:
         #no index.qmd here
         part: one.qmd
         chapters:
              - intro.qmd
              - literature.qmd
         part: two.qmd
              - hypothesis.qmd
              - experiments.qmd
              - results.qmd
              - conclusion.qmd
```
in `_extension.yml` I would populate the `book` metadata.

**Comment by dragonstyle** [comment, marked as answer]
> Margin reference style

Right now there isn't a way to customize how we place items in the margin (aside from altering the CSL which controls the format of the reference we place in the margin). If you're attempting to create links to the bibliography in the margin, would it be possible to simple place a citation in the margin (e.g. @ref-rahimi:2017 ) rather than the reference information itself?

> Sections

Does `number-sections: false` do what you're looking for? Aside from this customization, someone with a bit more LaTeX chops might be able to weight in...

> User Options

Right now, project extensions can't generate the book contents themselves (except in a hard coded way). You might be able to accomplish this with a project script (pre-processor):

https://quarto.org/docs/projects/scripts.html#pre-and-post-render
